src/mongodb/scores.controller.ts

```diff
@@ -1456,52 +1456,57 @@ export class ScoresController {
+            const decoded = audioFile.toString('base64');
+  
+            // Send Audio file to ASR to process and provide vector with char and score
+            let audioOutput = await this.scoresService.audioFileToAsrOutput(decoded, CreateLearnerProfileDto.language, CreateLearnerProfileDto['contentType']);
+  
+            asrOutDenoised = audioOutput.asrOutDenoisedOutput?.output || "";
+            asrOutBeforeDenoised = audioOutput.asrOutBeforeDenoised?.output || "";
+            pause_count = audioOutput.pause_count || 0;
+  
+            similarityDenoisedText = await this.scoresService.getTextSimilarity(originalText, asrOutDenoised[0]?.source || "");
+            similarityNonDenoisedText = await this.scoresService.getTextSimilarity(originalText, asrOutBeforeDenoised[0]?.source || "");
+  
+            if (similarityDenoisedText <= similarityNonDenoisedText) {
+              CreateLearnerProfileDto['output'] = asrOutBeforeDenoised;
+              DenoisedresponseText = await this.scoresService.processText(asrOutDenoised[0]?.source || "");
+              nonDenoisedresponseText = await this.scoresService.processText(asrOutBeforeDenoised[0]?.source || "");
+            } else {
+              CreateLearnerProfileDto['output'] = asrOutDenoised;
+              DenoisedresponseText = await this.scoresService.processText(asrOutDenoised[0]?.source || "");
+              nonDenoisedresponseText = await this.scoresService.processText(asrOutBeforeDenoised[0]?.source || "");
+            }
+  
+            if (CreateLearnerProfileDto.output[0].source === '') {
+              return response.status(HttpStatus.BAD_REQUEST).send({
+                status: 'error',
+                message:
+                  'Audio to Text functionality Responded Empty Response. Please check audio file or speak Loudly',
+              });
+            }
           }
 
-          if (CreateLearnerProfileDto.output[0].source === '') {
-            return response.status(HttpStatus.BAD_REQUEST).send({
-              status: 'error',
-              message:
-                'Audio to Text functionality Responded Empty Response. Please check audio file or speak Loudly',
-            });
-          }
+        responseText = await this.scoresService.processText(CreateLearnerProfileDto.output[0].source);
+        }
+        else {
+          responseText = CreateLearnerProfileDto.response_text;
+          pause_count = CreateLearnerProfileDto.pause_count;
         }
-
         // Get All hexcode for this selected language
         const tokenHexcodeDataArr = await this.scoresService.gethexcodeMapping(language);
-
-        responseText = await this.scoresService.processText(CreateLearnerProfileDto.output[0].source);
-
-        const textEvalMatrices = await this.scoresService.getTextMetrics(originalText, responseText, language, audioFile)
-
+        const textEvalMatrices = await this.scoresService.getTextMetrics(originalText, responseText, language)
```

Comment:
@DevendraPPatil You need to update this call parameters in all languages

Reply:
Yes Sure!

Reply:
I have checked It is already updated.